### PR TITLE
Query polish

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** #692 

**Proposed Changes:**

1. Update YAML types to reflect what's being done in practice
2. Update examples to be correct wrt spec

These are technically breaking changes, but I think they more accurately reflect the handful of implementations of this.

**PR Checklist:**

- [ ] This PR has **no** breaking changes.
- [X] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [X] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: api-spec/STAC-extensions.yaml [hunk 1]
```diff
@@ -1906,50 +1911,74 @@ components:
               description: >-
                 Find items with a property that is equal to the specified value.
                 For strings, a case-insensitive comparison must be performed.
+              oneOf:
```
Comment (m-mohr) [inline api-spec/STAC-extensions.yaml]:
```suggestion
              nullable: true
              oneOf:
```

File: api-spec/STAC-extensions.yaml [hunk 2]
```diff
@@ -1906,50 +1911,74 @@ components:
...
+                - type: string
+                - type: number
+                - type: boolean
             neq:
               description: >-
                 Find items that *don't* contain the specified value. For
                 strings, a case-insensitive comparison must be performed.
+              oneOf:
```
Comment (m-mohr) [inline api-spec/STAC-extensions.yaml]:
```suggestion
              nullable: true
              oneOf:
```

**Comment by m-mohr** [comment]
Looks good, but eq/neq schould support null, too.

**Comment by philvarner** [comment]
Updated nullable support

**Comment by m-mohr** [comment]
> Updated nullable support

Yes, thanks, but you forgot to run npm run generate-all, right?

**Comment by philvarner** [comment]
@m-mohr I did, update now

**Pull request merged**